easybuild/framework/easyblock.py

```diff
@@ -3038,20 +3038,42 @@ def run_all_steps(self, run_test_cases):
 
         print_msg("building and installing %s..." % self.full_mod_name, log=self.log, silent=self.silent)
         trace_msg("installation prefix: %s" % self.installdir)
-        try:
-            for (step_name, descr, step_methods, skippable) in steps:
-                if self._skip_step(step_name, skippable):
-                    print_msg("%s [skipped]" % descr, log=self.log, silent=self.silent)
-                else:
-                    if self.dry_run:
-                        self.dry_run_msg("%s... [DRY RUN]\n", descr)
-                    else:
-                        print_msg("%s..." % descr, log=self.log, silent=self.silent)
-                    self.current_step = step_name
-                    self.run_step(step_name, step_methods)
 
-        except StopException:
-            pass
+        lockpath = build_option('lockpath') or os.path.join(install_path('software'), '.locks')
+        if not os.path.exists(lockpath):
+            mkdir(lockpath, parents=True)
+        lockfile_name = os.path.join(lockpath, ".%s.lock" % self.installdir.replace('/', '_'))
+        if os.path.exists(lockfile_name):
```

Comment:
You do know that the only thing that works reliably as a lock is a directory. A file doesn't really work as a lock.
Even if you do check for existing lockfile and then create if it isn't there, you can't be sure it was you who created it.

So a file system lock entity should be a directory, not a file. mkdir is atomic, it either succeeeds becuse it didn't exist or it fails because someone else was fast enough to make it just between your exist-check and your mkdir.